src/Features/CSharp/Portable/QuickInfo/CSharpSyntacticQuickInfoProvider.cs

```diff
@@ -28,20 +30,38 @@ public CSharpSyntacticQuickInfoProvider()
         protected override Task<QuickInfoItem?> BuildQuickInfoAsync(
             QuickInfoContext context,
             SyntaxToken token)
-            => Task.FromResult(BuildQuickInfo(token));
+            => Task.FromResult(BuildQuickInfo(token, context.CancellationToken));
 
         protected override Task<QuickInfoItem?> BuildQuickInfoAsync(
             CommonQuickInfoContext context,
             SyntaxToken token)
-            => Task.FromResult(BuildQuickInfo(token));
+            => Task.FromResult(BuildQuickInfo(token, context.CancellationToken));
 
-        private static QuickInfoItem? BuildQuickInfo(SyntaxToken token)
+        private static QuickInfoItem? BuildQuickInfo(SyntaxToken token, CancellationToken cancellationToken)
         {
-            if (token.Kind() != SyntaxKind.CloseBraceToken)
+            switch (token.Kind())
             {
-                return null;
+                case SyntaxKind.CloseBraceToken:
+                    return BuildQuickInfoCloseBrace(token);
+                case SyntaxKind.HashToken:
+                case SyntaxKind.EndRegionKeyword:
+                case SyntaxKind.EndIfKeyword:
+                case SyntaxKind.ElseKeyword:
+                case SyntaxKind.ElifKeyword:
+                case SyntaxKind.EndOfDirectiveToken:
+                    return BuildQuickInfoDirectives(token, cancellationToken);
+                case SyntaxKind.EndOfFileToken:
+                    var endRegionKeyword = token.LeadingTrivia.LastOrDefault().GetStructure()?.ChildTokens().FirstOrDefault(t => t.IsKind(SyntaxKind.EndRegionKeyword));
+                    return endRegionKeyword.HasValue
+                        ? BuildQuickInfoDirectives(endRegionKeyword.Value, cancellationToken)
+                        : null;
```

Comment:
I think this should be dropped, because it just complicates things without much gain.

Reply:
what cases is it needed for?

Reply:
Just an `#endregion$$` with EOF following.

Reply:
I could take a commit and remove the test cases that fail after the removal

Reply:
I added commit 6b670383491f9e0fc4ef9788e1688e3463c5fd05
There I removed the EOF handling and the removed test cases failed after that changed. I can revert the change if you think that the cases are important to support.